org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/InferenceContext18.java

```diff
@@ -1794,7 +1794,7 @@ public ReferenceBinding inferRecordPatternParameterization(
 		if (!(typeBinding instanceof ReferenceBinding))
 			return null; // should not happen.
 		// 1.If T is not downcast convertible (5.5) to the raw type R, inference fails.
-		if (!typeBinding.isCompatibleWith(candidateT))
+		if (!isCompatibleWith(candidateT, typeBinding))
```

Comment:
We have a specific understanding of `isCompatibleWith`, but the new method seems to do something different. 
Does the implementation conform to JLS 5.5? 
We didn't yet have an implementation that does this?
If the method is needed indeed, perhaps `isDowncastConvertibleTo` is a better name?